### PR TITLE
UX: enhances chat copy features

### PR DESCRIPTION
**Pull request by jjaffeux**
- Allows to copy quotes from mobile
- Allows to copy text of a message from mobile

Screenshot 2023-10-04 at 11 28 06

- Allows to select messages by clicking on it when selection has started

Note this commit is also now using toasts to show a confirmation of copy, and refactors  system specs helpers concerning secondary actions.

Screenshot 2023-10-04 at 11 50 43

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/chat-message.gjs [hunk 1]
```diff
@@ -278,10 +280,27 @@ export default class ChatMessage extends Component {
     recursiveExpand(this.args.message);
   }
 
+  @action
+  toggleCheckIfPossible() {
+    if (!this.pane.selectingMessages) {
+      return;
+    }
+
+    if (event.shiftKey) {
```
[the remark this answers is not in the document]
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat-message.gjs:289]: yes indeed that's a mistake thanks

File: plugins/chat/assets/javascripts/discourse/lib/chat-message-interactor.js [hunk 2]
```diff
@@ -166,6 +167,14 @@ export default class ChatMessageInteractor {
       icon: "link",
     });
 
+    if (this.site.mobileView) {
+      buttons.push({
```
[the remark this answers is not in the document]
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/lib/chat-message-interactor.js:171]: It copies the raw

File: plugins/chat/assets/javascripts/discourse/lib/chat-message-interactor.js [hunk 3]
```diff
@@ -251,6 +268,10 @@ export default class ChatMessageInteractor {
 
     url = url.indexOf("/") === 0 ? protocol + "//" + host + url : url;
     clipboardCopy(url);
+    this.toasts.success({
+      duration: 3000,
```
[the remark this answers is not in the document]
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/lib/chat-message-interactor.js:272]: Yes agreed, it's not necessary, there's a default, but I though in this case it was too long. Didn't want to act globally yet, but I think we might want to reduce the default one for sure.

**Comment by jjaffeux** [comment]
Thanks!

**Review by markvanlan — APPROVED** [review]

**Pull request merged**